server/src/repositories/Store/poRepository.ts

```diff
@@ -0,0 +1,187 @@
+import { knex } from "../../db";
+import { PurchaseOrderMappper } from "../../mappers/store/PurchaseOrderMapper";
+import { InventoryItem } from "../../models/store/inventoryItem";
+import { PurchaseOrder } from "../../models/store/purchaseOrder";
+import { PurchaseOrderItemInput } from "../../models/store/purchaseOrderItemInput";
+
+export interface IPurchaseOrderRepo {
+  getAllPOs(): Promise<PurchaseOrder[]>;
+  getPOById(poId: number): Promise<PurchaseOrder>;
+  createNewPO(creatorId: number, expectedDeliveryDate: Date, items: PurchaseOrderItemInput[], attachments: string[]): Promise<PurchaseOrder>;
+  addItemToPO(item: InventoryItem, poId: number, count: number): Promise<PurchaseOrder>;
+  addItemsToPO(poId: number, items: PurchaseOrderItemInput[]): Promise<PurchaseOrder>;
+  addAttachmentToPO(attachment: string, poId: number): Promise<PurchaseOrder>;
+  addAttachmentsToPO(attachments: string[], poId: number): Promise<PurchaseOrder>;
+  removeAttachmentFromPO(attachment: string, poId: number): Promise<PurchaseOrder>;
+  removeItemFromPO(itemId: number, poId: number): Promise<PurchaseOrder>;
+  updateItemCountInPO(itemId: number, count: number, poId: number): Promise<PurchaseOrder>;
+  deletePOById(poId: number): Promise<void>;
+}
+
+
+export class PurchaseOrderRepo implements IPurchaseOrderRepo {
+
+  private queryBuilder
+
+  constructor(queryBuilder?: any) {
+    this.queryBuilder = queryBuilder || knex
+  }
+
+  public async getAllPOs(): Promise<PurchaseOrder[]> {
```

Comment:
Similarly, I don't really understand why these functions need to be part of a class, rather than just a series of exported functions. Is it just so we can do dependency injection on `queryBuilder`?

Reply:
the `queryBuilder` parameter is for transactions.  I've been using a class for the repo so you could initialize a repo with a transaction object and then all performed methods would be apart of that one transaction. I was running into some limitations doing it this way though and was considering having an extra parameter on the methods for passing in an optional transaction object...